src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -1473,14 +1777,17 @@ private void onNetworkWindow(
             }
 
             final int slotCapacity = bufferPool.slotCapacity();
-            final int sharedReplyCredit = Math.min(slotCapacity, replyBudget - encodeSlotOffset - sharedReplyBudget);
             sessions.values().forEach(s -> minimum.value = Math.min(s.remoteIncomingWindow, minimum.value));
 
-            if (sharedReplyCredit > 0)
+            final int replySharedBudgetMax = sessions.values().size() > 0 ?
+                (int) Math.min(minimum.value * encodeMaxFrameSize, replyBudget) : replyBudget;
+            final int replySharedCredit = replySharedBudgetMax - Math.max(replySharedBudget, 0) - Math.max(encodeSlotOffset, 0);
+
+            if (replySharedCredit > 0)
             {
-                final long replySharedBudgetPrevious = creditor.credit(traceId, replyBudgetIndex, sharedReplyCredit);
-                sharedReplyBudget = Math.min(minimum.value * initialMaxFrameSize, replyBudget);
+                final long replySharedBudgetPrevious = creditor.credit(traceId, replyBudgetIndex, replySharedCredit);
 
+                replySharedBudget += replySharedCredit;
```

Comment:
Take a look at recent changes to `http2` reply shared budget and make adjustments to above accordingly.